### ISSUE TITLE
Remove unnecessary manifest permissions

### ISSUE THREAD

**Issue by rhashimoto** [MetaMask/metamask-extension#809]
An extension should preferably enable only the minimum set of permissions.

app/manifest.json contains the `clipboardRead` permission, which enables an extension to read from the clipboard. I could not find any usage of `document.execCommand('paste')` in the code, and it doesn't seem like MetaMask should require free access to the clipboard contents.

app/manifest.json also contains the `tabs` permission, which is not needed to use the `chrome.tabs` or `chrome.windows` APIs, but only to enable the `url`, `title`, or `favIconUrl` properties of a `Tab` instance. I didn't find any such usage, but this was a more difficult search that I am less confident about. Does MetaMask need this permission anywhere?

**Comment by flyswatter** [comment]
I think you may be right. I enabled those permissions because Opera would not install MetaMask without them enabled, so I assumed they were required, but our clipboard usage may not actually require that permission.

- Dan

**Comment by rhashimoto** [comment]
From the point of view of a user, I would really prefer not to worry about an extension that could be reading my clipboard.

I realize that MetaMask already must be trusted to run content scripts on every page I visit, but that's a separate issue from minimizing the permission set.

**Comment by flyswatter** [comment]
Right, I agree with you.

We'll do some investigation, and probably redact this permission soon.

**Comment by flyswatter** [comment]
The only reason I'm not removing this with urgency, is because when we *did* add this permission, it pinged all our users for extra permissions, and it scared some people away. If it *does* turn out we need this permission, I don't want us to remove it just to add it again, scaring our users all over again.

Since this permission was originally added for Opera support, I think this issue is somewhat dependent on #698, so I'll nudge that along (I haven't heard from them in a while).

**Comment by flyswatter** [comment]
Maybe we should only include those permissions on Opera builds.

**Comment by flyswatter** [comment]
We could just remove these permissions proactively, then re-add them to an Opera build later.

**Comment by spundot** [comment]
The current version on Chrome Web Store (v3.13.8) still requests the copy/paste permission, as noted here: https://github.com/MetaMask/metamask-extension/pull/965#issuecomment-357444318

Can this be reopened?

**Comment by danfinlay** [comment]
Reopened with a more specific title.

Closing this would require both removing unneeded keys, and demonstrating that functionality is not lost in any of the browsers we support:

- Chrome
- Firefox
- Opera
- Edge (soon)
- Brave (should work as long as Chrome does)

**Comment by bdresser** [comment]
Could also add some things as "optional permissions" which we ask for only when a user tries to use a particular feature. From https://github.com/MetaMask/metamask-extension/issues/5196


>As described here:
https://developer.chrome.com/apps/permissions

> There is an optional permissions field that can ask permissions from the user when they decide to use a feature. In the spirit of fostering trust with our users, I think that this should be an easy fix--literally just modifying the manifest to move permissions, such as notifications and copy/paste, into the optional_permissions field

**Comment by bdresser** [comment]
<img width="466" alt="screen shot 2019-03-06 at 3 46 45 pm" src="https://user-images.githubusercontent.com/1016190/53921894-11520180-4027-11e9-8a8c-d4ead65dbe2f.png">

**Comment by bdresser** [comment]
Might be possible to dynamic requesting of permissions -- only ask when we need it. Check back in once https://github.com/MetaMask/metamask-extension/pull/6272 is merged

Also would be nice to tighten the specific "modify data you copy and paste" permission since we don't need to modify, just add to the clipboard.

**Comment by bdresser** [comment]
We should make sure this doesn't prompt the user to agree to a new/other set of permissions

**Comment by whymarrh** [comment]
We've merged #6654 (linked in the timeline above as well) which removes the `clipboardWrite` permission. That, I believe, was the only unnecessary permission we had listed.
